presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergSessionProperties.java

```diff
@@ -0,0 +1,139 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.hive.HiveClientConfig;
+import com.facebook.presto.hive.HiveCompressionCodec;
+import com.facebook.presto.hive.ParquetFileWriterConfig;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.session.PropertyMetadata;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.common.type.VarcharType.createUnboundedVarcharType;
+import static com.facebook.presto.spi.session.PropertyMetadata.booleanProperty;
+
+public final class IcebergSessionProperties
+{
+    private static final String COMPRESSION_CODEC = "compression_codec";
+    private static final String PARQUET_FAIL_WITH_CORRUPTED_STATISTICS = "parquet_fail_with_corrupted_statistics";
+    private static final String PARQUET_MAX_READ_BLOCK_SIZE = "parquet_max_read_block_size";
+    private static final String PARQUET_WRITER_BLOCK_SIZE = "parquet_writer_block_size";
+    private static final String PARQUET_WRITER_PAGE_SIZE = "parquet_writer_page_size";
+    private static final String PARQUET_USE_COLUMN_NAMES = "parquet_use_column_names";
+    private static final String PARQUET_BATCH_READ_OPTIMIZATION_ENABLED = "parquet_batch_read_optimization_enabled";
+    private static final String PARQUET_BATCH_READER_VERIFICATION_ENABLED = "parquet_batch_reader_verification_enabled";
+    private final List<PropertyMetadata<?>> sessionProperties;
+
+    @Inject
+    public IcebergSessionProperties(
+            IcebergConfig icebergConfig,
+            HiveClientConfig hiveClientConfig,
+            ParquetFileWriterConfig parquetFileWriterConfig)
+    {
+        sessionProperties = ImmutableList.of(
+                new PropertyMetadata<>(
+                        COMPRESSION_CODEC,
+                        "The compression codec to use when writing files",
+                        VARCHAR,
+                        HiveCompressionCodec.class,
+                        hiveClientConfig.getCompressionCodec(),
+                        false,
+                        value -> HiveCompressionCodec.valueOf(((String) value).toUpperCase()),
+                        HiveCompressionCodec::name),
+                booleanProperty(
+                        PARQUET_FAIL_WITH_CORRUPTED_STATISTICS,
+                        "Parquet: Fail when scanning Parquet files with corrupted statistics",
+                        hiveClientConfig.isFailOnCorruptedParquetStatistics(),
+                        false),
+                booleanProperty(
+                        PARQUET_USE_COLUMN_NAMES,
```

Comment:
Do we really use this in iceberg?

Reply:
Yes, it is used in the `createDataPageSource` function in the `IcebergPageSourceProvider.java` file~